activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -183,7 +182,7 @@ def normalize_key(key, options)
           key = super.dup
           key = key.force_encoding(Encoding::ASCII_8BIT)
           key = key.gsub(ESCAPE_KEY_CHARS) { |match| "%#{match.getbyte(0).to_s(16).upcase}" }
-          key = "#{key[0, 213]}:md5:#{Digest::MD5.hexdigest(key)}" if key.size > 250
+          key = "#{key[0, 213]}:md5:#{ActiveSupport::Digest.hexdigest(key)}" if key.size > 250
```

Comment:
Won't this need to be dynamic now the hash is not hardcoded anymore? `":md5:#{Digest::MD5.hexdigest(key)}" is 37 characters, but with SHA1 it's 45 so it'll go over the 250 byte limit.

Reply:
I saw that, but didn't think it was necessary to change. My understanding is that this is only a hint in the key, and doesn't effect the functionality in any way. TBH, I'm not sure why use md5 there at all.

Reply:
Memcached keys can't be longer than 250 characters:

https://github.com/memcached/memcached/blob/master/doc/protocol.txt

> Data stored by memcached is identified with the help of a key. A key is a text string which should uniquely identify the data for clients that are interested in storing and retrieving it.  Currently the length limit of a key is set at 250 characters (of course, normally clients wouldn't need to use such long keys); the key must not include control characters or whitespace.

We should truncate the output of `ActiveSupport::Digest.hexdigest` to 32 characters.

Reply:
Or what Dalli does - adjust key truncation according to digest size. I submitted a fix #45245